mithril-aggregator/src/certifier_service.rs

```diff
@@ -673,4 +709,53 @@ mod tests {
             .unwrap();
         assert!(create_certificate_result.is_none());
     }
+
+    #[tokio::test]
+    async fn test_epoch_gap_certificate_chain() {
+        let builder = MithrilFixtureBuilder::default();
+        let certifier_service = setup_certifier_service(&builder.build(), &[]).await;
+        let mut certificate = fake_data::certificate("one".to_string());
+        certificate.genesis_signature = "whatever".to_string();
+        certifier_service
+            .certificate_repository
+            .create_certificate(certificate)
+            .await
+            .unwrap();
+        let epoch = Epoch(12);
+        let error = certifier_service
+            .verify_certificate_chain(epoch)
+            .await
+            .unwrap_err();
+
+        if let Some(err) = error.downcast_ref::<CertifierServiceError>() {
+            assert!(matches!(err, CertifierServiceError::CertificateEpochGap(ep) if *ep == epoch));
+        } else {
+            panic!("Unexpected error {error:?}");
+        }
+    }
+
+    #[tokio::test]
+    async fn test_epoch_gap_certificate_chain_ok() {
+        let builder = MithrilFixtureBuilder::default();
+        let certifier_service = setup_certifier_service(&builder.build(), &[]).await;
+        let mut certificate = fake_data::certificate("one".to_string());
```

Comment:
Same as the previous comment: use `fake_data::genesis_certificate`  and specify the certificate epoch :).